packages/backend/src/controllers/bills/bills.controller.ts

```diff
@@ -0,0 +1,191 @@
+        ).map((bill) => {
+          return this.billUtil.covertBillDocumentToResponseDTO(
+            bill,
+            this.userStoreService,
+          );
+        }),
+      ),
+    };
+  }
+
+  @Post()
+  @ApiOperation({ summary: 'Create a new bill resource' })
+  @ApiCreatedResponse({
+    description: 'Bill created successfully',
+    type: BillResponseDto,
+  })
+  async createBill(
+    @Body() createBillDto: CreateBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const owner = await this.userStoreService.findOneByFirebaseId(user.uid);
+    const users = await Promise.all(
+      createBillDto.users.map(async (u) => {
+        return {
+          id: (await this.userStoreService.findOneByFirebaseId(user.uid))._id,
+          amount: u.amount,
+          paid: u.paid,
+          proof: u.proof,
+        };
+      }),
+    );
+    const billModel: BillModel = {
+      name: createBillDto.name,
+      description: createBillDto.description,
+      house: owner.house,
+      owner: owner._id,
+      due: new Date(createBillDto.due),
+      users: users,
+    };
+    const bill = await this.billStoreService.create(billModel);
+    return {
+      name: bill.name,
+      description: bill.description,
+      owner: owner.firebaseId,
+      due: createBillDto.due,
+      users: createBillDto.users,
+    };
+  }
+
+  @Put(':id')
+  @ApiOperation({ summary: 'Update a bill.' })
+  @ApiOkResponse({
+    description: 'Bill updated successfully',
+    type: BillResponseDto,
+  })
+  @ApiUnauthorizedResponse({ description: 'not the bill owner' })
+  async updateBill(
+    @Param() idObject: string,
+    @Body() updateBillDto: UpdateBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const billId = idObject['id'];
+    const bill = await this.billStoreService.findOne(billId);
+    const userObject = await this.userStoreService.findOneByFirebaseId(
+      user.uid,
+    );
+
+    if (bill.owner.valueOf() === userObject._id.valueOf()) {
+      const billDocument = await this.billStoreService.update(
+        bill._id,
+        updateBillDto,
+      );
+      return this.billUtil.covertBillDocumentToResponseDTO(
+        billDocument,
+        this.userStoreService,
+      );
+    } else
+      throw new HttpException('not the bill owner', HttpStatus.UNAUTHORIZED);
+  }
+
+  @Put(':id/payment')
+  @ApiOperation({ summary: 'Mark a user as having paid or unpaid a bill.' })
+  @ApiOkResponse({
+    description: 'Bill marked successfully',
+    type: BillResponseDto,
+  })
+  async updateBillPayment(
+    @Param() idObject: string,
+    @Body() payBillDto: PayBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const billId = idObject['id'];
+    const bill = await this.billStoreService.findOne(billId);
+    const userObject = await this.userStoreService.findOneByFirebaseId(
+      user.uid,
+    );
+    bill.users.map((u) => {
```

Comment:
This should probably be foreach instead of map

Reply:
yes currently it is doing nothing because nothing is being done to the returned array

Reply:
but it shouldnt be forEach. just make `bill.users` equal to the returned array from map

Reply:
forEach seems to be fine to me since it passes the reference of the object to the anonymous function therefore you can 
 modify the object during iteration. If the array contained primitive values you would be right. 
 In fact map would also work in this scenario but it doesn't make sense to use it.

Reply:
Okay great, I am not a JS expert, but yeah also long as we change to forEach so that the `bill.users` will actually get updated in the db.